plugins/module_utils/network/sonic/config/sflow/sflow.py

```diff
@@ -0,0 +1,522 @@
+        if "sampling_rate" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/sampling-rate", "method": "DELETE"})
+        if "collectors" in to_delete:
+            have_collectors_dict = {(collector["address"], collector["network_instance"], collector["port"]): collector for collector in have["collectors"]}
+            to_delete_collectors_dict = {(collector["address"], collector["network_instance"], collector["port"]): collector
+                                         for collector in to_delete["collectors"]}
+            if have_collectors_dict.keys() == to_delete_collectors_dict.keys():
+                requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/collectors", "method": "DELETE"})
+            else:
+                for collector in to_delete["collectors"]:
+                    requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/collectors/collector=" +
+                                    collector["address"] + "," + str(collector["port"]) + "," +
+                                    collector["network_instance"], "method": "DELETE"})
+        if "interfaces" in to_delete:
+            for interface in to_delete["interfaces"]:
+                requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/interfaces/interface=" + interface["name"], "method": "DELETE"})
+        return requests
+
+    def fill_defaults(self, config):
+        '''modifies the given original config object to add sflow default values that are missing. returns the config for chaining purposes'''
+        if "enabled" not in config:
+            config["enabled"] = False
+        return config
+
+    def find_substitution_deletes(self, remove_diff, introduced_diff):
+        '''specifically for overridden and replaced states, finds and builds collection of which config settings need to be deleted and without anything that is
+        getting replaced with new values. `get_diff` will return collection of both things that need to be deleted and things that will have new values.'''
+        result = {}
+        if "agent" in remove_diff and "agent" not in introduced_diff:
+            result["agent"] = remove_diff["agent"]
+        if "enabled" in remove_diff and "enabled" not in introduced_diff:
+            result["enabled"] = remove_diff["enabled"]
+        if "polling_interval" in remove_diff and "polling_interval" not in introduced_diff:
+            result["polling_interval"] = remove_diff["polling_interval"]
+        if "sampling_rate" in remove_diff and "sampling_rate" not in introduced_diff:
+            result["sampling_rate"] = remove_diff["sampling_rate"]
+        if "collectors" in remove_diff:
+            result["collectors"] = remove_diff["collectors"]
+        if "interfaces" in remove_diff:
+            result["interfaces"] = remove_diff["interfaces"]
+        return result
+
+    def prep_merge_diff(self, diff, want):
```

Comment:
It looks like this function and the build_ref_dict and find_item_key functions are not needed after making the suggested changes for "find_substitution_deletes".